packages/serverless/src/awsservices.ts

```diff
@@ -15,62 +16,68 @@ interface AWSService {
   serviceIdentifier: string;
 }
 
-/** AWS service requests tracking */
-export class AWSServices implements Integration {
-  /**
-   * @inheritDoc
-   */
-  public static id: string = 'AWSServices';
+const INTEGRATION_NAME = 'AWSServices';
 
-  /**
-   * @inheritDoc
-   */
-  public name: string;
+const SETUP_CLIENTS = new WeakMap<Client, boolean>();
 
-  private readonly _optional: boolean;
-
-  public constructor(options: { optional?: boolean } = {}) {
-    this.name = AWSServices.id;
+const _awsServicesIntegration = ((options: { optional?: boolean } = {}) => {
+  const optional = options.optional || false;
+  return {
+    name: INTEGRATION_NAME,
+    setupOnce() {
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const awsModule = require('aws-sdk/global') as typeof AWS;
+        fill(awsModule.Service.prototype, 'makeRequest', wrapMakeRequest);
+      } catch (e) {
+        if (!optional) {
+          throw e;
+        }
+      }
+    },
+    setup(client) {
+      SETUP_CLIENTS.set(client, true);
+    },
+  };
+}) satisfies IntegrationFn;
 
-    this._optional = options.optional || false;
-  }
+export const awsServicesIntegration = defineIntegration(_awsServicesIntegration);
 
-  /**
-   * @inheritDoc
-   */
-  public setupOnce(): void {
-    try {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const awsModule = require('aws-sdk/global') as typeof AWS;
-      fill(awsModule.Service.prototype, 'makeRequest', wrapMakeRequest);
-    } catch (e) {
-      if (!this._optional) {
-        throw e;
-      }
-    }
-  }
-}
+/**
+ * AWS Service Request Tracking.
+ *
+ * @deprecated Use `awsServicesIntegration()` instead.
+ */
+// eslint-disable-next-line deprecation/deprecation
+export const AWSServices = convertIntegrationFnToClass(
+  INTEGRATION_NAME,
+  awsServicesIntegration,
+) as IntegrationClass<Integration>;
```

Comment:
let's also add:

```ts
export type AWSServices = typeof AWSServices;
```

right away here (same for the gcp stuff)!

Reply:
do we have to also export the type from the index?

Reply:
otherwise this does not work anymore (it does not work with the stuff I migrated):

```ts
import {AWSServices} from '@sentry/serverless';

let a: AWSServices | undefined;
// instead you need to do this:
let b: typeof AWSServices | undefined;
```

This is very unlikely to affect somebody, but for consistency we should do that (and also for all other integrations, just didn't get to it yet...)